memcache/memcache_test.go

```diff
@@ -301,3 +302,45 @@ func BenchmarkScanGetResponseLine(b *testing.B) {
 		}
 	}
 }
+
+func BenchmarkParseGetResponse(b *testing.B) {
+	response := strings.NewReader("VALUE foobar1234 0 5 1234\r\nhello\r\nEND\r\n")
+	c := &Client{
+		Pool: newTestPool(5),
+	}
+	var reader = bufio.NewReader(response)
+	var err error
+	for i := 0; i < b.N; i++ {
+		err = c.parseGetResponse(reader, func(it *Item) {
+			c.Pool.Put(&it.Value)
+		})
+		if err != nil {
+			b.Fatal(err)
+		}
+		response.Seek(0, 0)
+		reader.Reset(response)
+	}
+}
+
+type testPool struct {
+	pool sync.Pool
+}
+
+func newTestPool(dataSize int) BytesPool {
+	return &testPool{
+		pool: sync.Pool{
+			New: func() interface{} {
+				b := make([]byte, 0, dataSize+2)
```

Comment:
why +2 ?

Reply:
Because the buffers are allocated with a capacity of 2 more than the data size to accommodate for the terminating `\r\n` (not included in the size metadata in the response). So for the case of our test pool, it will need to provide buffers than are 2 bigger than the data size.
https://github.com/grafana/gomemcache/blob/3a2f55c465086fc4dc12a29b58e6b59882a3f78d/memcache/memcache.go#L510

Reply:
I see you moved this up to the call, which seems more sensible to me.

Reply:
Ok, I moved it up.